# $\alpha$=0.1

In [1]:
import torch
import torchvision.transforms as transforms              
from torchvision.datasets import CIFAR100      
from torch.utils.data import DataLoader
from src.inception_cifar100 import inceptionv3
from src.temperature_scaling import ModelWithTemperature
import numpy as np
from src.aps import aps_test


device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

dict_path = "C:\\Users\\jiayang\\ipynb\\trainedModel\\Inception_CIFAR100.pth"
model = inceptionv3()
model.load_state_dict(torch.load(dict_path, map_location=device, weights_only=True))
model.to(device)

# preprocess the images from CIFAR100
data_transform = transforms.Compose([
    transforms.ToTensor(),         
    transforms.Normalize((0.5071, 0.4865, 0.4409), (0.2673, 0.2564, 0.2762))  
])

# load data set from CIFAR100
dataset = CIFAR100(root="../../data", train=False, download=True,transform=data_transform)

# Temperature Scaling
temp_scal_loader = DataLoader(dataset, batch_size=32, shuffle=True)
model = ModelWithTemperature(model, temperature=0.5).to(device)
model.set_temperature(temp_scal_loader)
model.eval()

aps_test(model, dataset, device, num_runs=10, alpha=0.1)

Files already downloaded and verified
Before temperature - NLL: 1.388, ECE: 0.119
Optimal temperature: 0.633
After temperature - NLL: 1.879, ECE: 0.201
APS Classification, Start!

Running experiment 1/10...
q_hat = 0.9968848466873169
Total set size: 17467
Total coverage sets: 4543
Total samples amount: 5000
Average Prediction Set Size After APS in runs 1: 3.4934
Average Coverage Rate in runs 1: 0.9086

Running experiment 2/10...
q_hat = 0.9966991186141968
Total set size: 17636
Total coverage sets: 4518
Total samples amount: 5000
Average Prediction Set Size After APS in runs 2: 3.5272
Average Coverage Rate in runs 2: 0.9036

Running experiment 3/10...
q_hat = 0.996415787935257
Total set size: 17276
Total coverage sets: 4495
Total samples amount: 5000
Average Prediction Set Size After APS in runs 3: 3.4552
Average Coverage Rate in runs 3: 0.899

Running experiment 4/10...
q_hat = 0.9959723591804505
Total set size: 16471
Total coverage sets: 4477
Total samples amount: 5000
Average Predict

# $\alpha$=0.2

In [2]:
aps_test(model, dataset, device, num_runs=10, alpha=0.2)

Files already downloaded and verified
Before temperature - NLL: 1.387, ECE: 0.123
Optimal temperature: 0.633
After temperature - NLL: 1.877, ECE: 0.206
APS Classification, Start!

Running experiment 1/10...
q_hat = 0.9648807406425476
Total set size: 9241
Total coverage sets: 4036
Total samples amount: 5000
Average Prediction Set Size After APS in runs 1: 1.8482
Average Coverage Rate in runs 1: 0.8072

Running experiment 2/10...
q_hat = 0.9597337484359741
Total set size: 9096
Total coverage sets: 3995
Total samples amount: 5000
Average Prediction Set Size After APS in runs 2: 1.8192
Average Coverage Rate in runs 2: 0.799

Running experiment 3/10...
q_hat = 0.9607915282249451
Total set size: 9085
Total coverage sets: 3999
Total samples amount: 5000
Average Prediction Set Size After APS in runs 3: 1.817
Average Coverage Rate in runs 3: 0.7998

Running experiment 4/10...
q_hat = 0.9599417567253112
Total set size: 8931
Total coverage sets: 3971
Total samples amount: 5000
Average Prediction 

# $\alpha$=0.05

In [2]:
aps_test(model, dataset, device, num_runs=10, alpha=0.05)

APS Classification, Start!

Running experiment 1/10...
q_hat = 0.9997429937124253
Total set size: 31860
Total coverage sets: 4768
Total samples amount: 5000
Average Prediction Set Size After APS in runs 1: 6.372
Average Coverage Rate in runs 1: 0.9536

Running experiment 2/10...
q_hat = 0.9997072726488113
Total set size: 31847
Total coverage sets: 4755
Total samples amount: 5000
Average Prediction Set Size After APS in runs 2: 6.3694
Average Coverage Rate in runs 2: 0.951

Running experiment 3/10...
q_hat = 0.9996629059314728
Total set size: 30666
Total coverage sets: 4733
Total samples amount: 5000
Average Prediction Set Size After APS in runs 3: 6.1332
Average Coverage Rate in runs 3: 0.9466

Running experiment 4/10...
q_hat = 0.9997211992740631
Total set size: 31495
Total coverage sets: 4761
Total samples amount: 5000
Average Prediction Set Size After APS in runs 4: 6.299
Average Coverage Rate in runs 4: 0.9522

Running experiment 5/10...
q_hat = 0.9996899694204331
Total set size: 3

#  Result
$\alpha$=0.1  
From the above test, following results can be collected :
- Final Average Prediction Set Size: 3.46
- Final Average Coverage: 90.15%   

$\alpha$=0.3  
From the above test, following results can be collected :
- Final Average Prediction Set Size: 1.29
- Final Average Coverage: 69.59%  

$\alpha$=0.2  
From the above test, following results can be collected :
- Final Average Prediction Set Size: 1.80
- Final Average Coverage: 79.79%  

$\alpha$=0.05  
From the above test, following results can be collected :
- Final Average Prediction Set Size: 6.23
- Final Average Coverage: 94.98%